be/src/vec/exec/format/parquet/parquet_column_convert.h

```diff
@@ -0,0 +1,623 @@
+            value = date_dict[date_value];
+        }
+
+        return Status::OK();
+    }
+};
+
+template <typename DecimalType, bool is_nullable, DecimalScaleParams::ScaleType ScaleType>
+class stringtodecimal : public ColumnConvert {
+public:
+    Status convert(const IColumn* src_col, IColumn* dst_col) override {
+        size_t rows = src_col->size();
+        if constexpr (is_nullable) {
+            convert_null(&src_col, &dst_col);
+        }
+        DecimalScaleParams& scale_params = _convert_params->decimal_scale;
+        auto buf = static_cast<const ColumnString*>(src_col)->get_chars().data();
+        auto& offset = static_cast<const ColumnString*>(src_col)->get_offsets();
+        dst_col->resize(_convert_params->start_idx + rows);
+
+        auto& data = static_cast<ColumnDecimal<DecimalType>*>(dst_col)->get_data();
+        for (int i = 0; i < rows; i++) {
+            int len = offset[i] - offset[i - 1];
+            // When Decimal in parquet is stored in byte arrays, binary and fixed,
+            // the unscaled number must be encoded as two's complement using big-endian byte order.
+            Int128 value = 0;
+            memcpy(reinterpret_cast<char*>(&value), buf + offset[i - 1], len);
+            value = BitUtil::big_endian_to_host(value);
+            value = value >> ((sizeof(value) - len) * 8);
+            if constexpr (ScaleType == DecimalScaleParams::SCALE_UP) {
+                value *= scale_params.scale_factor;
+            } else if constexpr (ScaleType == DecimalScaleParams::SCALE_DOWN) {
+                value /= scale_params.scale_factor;
+            } else if constexpr (ScaleType == DecimalScaleParams::NO_SCALE) {
+                // do nothing
+            } else {
+                LOG(FATAL) << "__builtin_unreachable";
+                __builtin_unreachable();
+            }
+            auto& v = reinterpret_cast<DecimalType&>(data[_convert_params->start_idx + i]);
+            v = (DecimalType)value;
+        }
+
+        return Status::OK();
+    }
+};
+template <typename NumberType, typename DecimalPhysicalType, bool is_nullable,
+          DecimalScaleParams::ScaleType ScaleType>
+class numbertodecimal : public ColumnConvert {
+public:
+    Status convert(const IColumn* src_col, IColumn* dst_col) override {
+        size_t rows = src_col->size();
+        if constexpr (is_nullable) {
+            convert_null(&src_col, &dst_col);
+        }
+        auto* src_data = static_cast<const ColumnVector<NumberType>*>(src_col)->get_data().data();
+        dst_col->resize(_convert_params->start_idx + rows);
+
+        DecimalScaleParams& scale_params = _convert_params->decimal_scale;
+        auto* data = static_cast<ColumnDecimal<Decimal<DecimalPhysicalType>>*>(dst_col)
+                             ->get_data()
+                             .data();
+
+        for (int i = 0; i < rows; i++) {
+            Int128 value = src_data[i];
```

Comment:
PR https://github.com/apache/doris/pull/25012 has optimized the performance of reading decimal, why still use int128 here?